File: 62-UniquePaths/62-UniquePaths.py
# Last updated: 4/29/2026, 10:35:37 AM
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        @cache
        def dp(row, col):
            if row == m-1 and col == n-1:
                return 1
            ways = 0
            if row < m - 1:
                ways += dp(row+1, col)
            if col < n - 1:
                ways += dp(row, col+1)
            return ways
            
        return dp(0, 0)

File: 62-UniquePaths/62-UniquePaths.js
// Last updated: 4/29/2026, 10:35:32 AM
/**
 * @param {number} m
 * @param {number} n
 * @return {number}
 */
var uniquePaths = function(m, n) {
    const matrix = [];
    for(let i = 0; i < m; i++) {
        const row = [];
        for(let j = 0; j < n; j++) {
            row.push(0);
        }
        matrix.push(row);
    }
    
    for(let i = 0; i < m; i++) {
        matrix[i][0] = 1;
    }
    for(let j = 0; j < n; j++) {
        matrix[0][j] = 1;
    }
    for(let i = 1; i < m; i++) {
        for(let j = 1; j < n; j++) {
            matrix[i][j] = matrix[i-1][j] + matrix[i][j-1];
        }
    }
    return matrix[m-1][n-1];
};